v2/src/logging_stacktrace.c

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdio.h>
+#include <string.h>
+#include <inttypes.h>
+#include <stddef.h>
+
+#ifdef WIN32
+#include "windows.h"
+#include "c_logging/get_thread_stack.h"
+#endif
+
+#include "c_logging/logging_stacktrace.h"
+
+// Thread-local storage for stack strings (compatibility with v1)
+XLOGGING_THREAD_LOCAL char stackAsString[STACK_MAX_CHARACTERS];
+XLOGGING_THREAD_LOCAL char formatWithStack[FORMAT_MAX_CHARACTERS];
+
+/*tries to get as much as possible from the stack filling destination*/
+void getStackAsString(char* destination, size_t destinationSize)
+{
+    /*Codes_SRS_LOGGING_STACKTRACE_01_002: [ If destination is NULL, getStackAsString shall return without doing anything. ]*/
+    /*Codes_SRS_LOGGING_STACKTRACE_01_003: [ If destinationSize is 0, getStackAsString shall return without doing anything. ]*/
+    if (destination == NULL || destinationSize == 0)
+    {
+        return;
+    }
```

Comment:
Interesting that you are returning in the middle of a function.  Would it be better to just have an else after this and then return at the bottom?

Reply:
Changed